### ISSUE TITLE
@CompiledJson annotation asking to give all parameters in Kotlin (default params not working)

### ISSUE THREAD

**Issue by Qudratillo** [ngs-doo/dsl-json#213]
Hello, compiler gives an error when i try to use `@CompiledJson` without parameters in Kotlin. However, in Java it works as expected.

Following code gives an error in Kotlin
```
@CompiledJson
class APIError {
}
```

I did some research on this issues and I found after release 1.8.0 this issue was born. Only change made to `CompiledJson` was adding `ElementType.METHOD` target. It was like

Before
```
@Target({ElementType.TYPE, ElementType.CONSTRUCTOR})
@Retention(RetentionPolicy.CLASS)
public @interface CompiledJson {
```

After
```
@Target({ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.METHOD})
@Retention(RetentionPolicy.CLASS)
public @interface CompiledJson {
```

Complier error output
```

**Comment by zapov** [comment]
I tried reproducing this, but it work in my test: https://github.com/ngs-doo/dsl-json/commit/2963fb37767292f8874e0b9eda3243951e8c8f63

You would need to provide more info, eg, Java version, Kotlin version, etc...
Ideally make a test like I did, but which fails

**Comment by Qudratillo** [comment]
@zapov Thank you for then answer

I downloaded the project and run tests, they executed fine. However, after changing kotlin version to 1.5.10, tests failed (same error I mentioned above). I change kotlin version to 1.4.32 and everything worked fine.
I conclude Kotlin version 1.5.0 introduced some changes that effected @CompiledJson annotation.

**Comment by Qudratillo** [comment]
<img width="1433" alt="Screen Shot 2021-06-30 at 16 23 15" src="https://user-images.githubusercontent.com/11327524/123952473-7c296280-d9bf-11eb-8008-2e32c659a8cb.png">

**Comment by re-thc** [comment]
Same error. It's definitely a Kotlin 1.5.0 issue. Likely some syntax change.

**Comment by zapov** [comment]
I don't use Kotlin much ;( It would be great if someone can look into this... maybe its some regression they did in the language

**Comment by Qudratillo** [comment]
```
@Target({ElementType.TYPE, ElementType.CONSTRUCTOR, ElementType.METHOD})
@Retention(RetentionPolicy.CLASS)
public @interface CompiledJson {
```

@zapov if you delete `ElementType.METHOD` target from CompiledJson, it must work.

**Comment by zapov** [comment]
But I can't do that since this annotation can be used for factories and builder to mark those methods.

**Comment by Qudratillo** [comment]
@zapov Yeah, I know. I said it to narrow the issue. I think we need to research on this target.

**Comment by zapov** [comment]
https://youtrack.jetbrains.com/issue/KT-48316
